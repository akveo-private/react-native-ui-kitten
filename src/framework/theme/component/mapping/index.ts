export * from './type';
export {
  MappingProvider as ThemeMappingProvider,
  Props as ThemeMappingProviderProps,
} from './mappingProvider.component';
