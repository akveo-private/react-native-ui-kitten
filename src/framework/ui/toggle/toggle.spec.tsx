import React from 'react';
import { TouchableOpacity } from 'react-native';
import {
  render,
  shallow,
  RenderAPI,
  fireEvent,
  waitForElement,
} from 'react-native-testing-library';
import { ReactTestInstance } from 'react-test-renderer';
import {
  styled,
  ApplicationProvider,
  ApplicationProviderProps,
} from '@kitten/theme';
import {
  Toggle as ToggleComponent,
  Props,
} from './toggle.component';
import { default as mapping } from '../common/mapping.json';
import { default as theme } from '../common/theme.json';

const Toggle = styled<Props>(ToggleComponent);

const Mock = (props?: Props): React.ReactElement<ApplicationProviderProps> => {
  return (
    <ApplicationProvider
      mapping={mapping}
      theme={theme}>
      <Toggle {...props} />
    </ApplicationProvider>
  );
};

const renderComponent = (props?: Props): RenderAPI => {
  return render(
    <Mock {...props}/>,
  );
};

describe('@toggle: matches snapshot', () => {

  it('default', () => {
    const component: RenderAPI = renderComponent();
    const { output } = shallow(component.getByType(ToggleComponent));

    expect(output).toMatchSnapshot();
  });

  it('checked', () => {
    const component: RenderAPI = renderComponent({ checked: true });
    const { output } = shallow(component.getByType(ToggleComponent));

    expect(output).toMatchSnapshot();
  });

  it('disabled', () => {
    const component: RenderAPI = renderComponent({ disabled: true });
    const { output } = shallow(component.getByType(ToggleComponent));

    expect(output).toMatchSnapshot();
  });

  it('checked disabled', () => {
    const component: RenderAPI = renderComponent({
      checked: true,
      disabled: true,
    });
    const { output } = shallow(component.getByType(ToggleComponent));

    expect(output).toMatchSnapshot();
  });

  it('active', async () => {
    const component: RenderAPI = renderComponent();

    fireEvent(component.getByType(TouchableOpacity), 'pressIn');

    const active: ReactTestInstance = await waitForElement(() => {
      return component.getByType(ToggleComponent);
    });
    const { output: activeOutput } = shallow(active);

    fireEvent(component.getByType(TouchableOpacity), 'pressOut');

    const inactive: ReactTestInstance = await waitForElement(() => {
      return component.getByType(ToggleComponent);
    });
    const { output: inactiveOutput } = shallow(inactive);

    expect(activeOutput).toMatchSnapshot();
    expect(inactiveOutput).toMatchSnapshot('default');
  });

  it('active checked', async () => {
    const component: RenderAPI = renderComponent({ checked: true });

    fireEvent(component.getByType(TouchableOpacity), 'pressIn');
    const active: ReactTestInstance = await waitForElement(() => {
      return component.getByType(ToggleComponent);
    });
    const { output: activeOutput } = shallow(active);

    fireEvent(component.getByType(TouchableOpacity), 'pressOut');

    const inactive: ReactTestInstance = await waitForElement(() => {
      return component.getByType(ToggleComponent);
    });
    const { output: inactiveOutput } = shallow(inactive);

    expect(activeOutput).toMatchSnapshot();
    expect(inactiveOutput).toMatchSnapshot('checked');
  });

});

describe('@toggle: component checks', () => {

  it('emits onChange', () => {
    const onChange = jest.fn();

    const component: RenderAPI = renderComponent({ onChange: onChange });

    fireEvent.press(component.getByType(TouchableOpacity));

    expect(onChange).toBeCalled();
  });

  it('checking of value direct', () => {
    let checked: boolean = false;
    const onChangeValue = (changed: boolean) => {
      checked = changed;
    };

    const component: RenderAPI = renderComponent({
      checked: checked,
      onChange: onChangeValue,
    });

    fireEvent.press(component.getByType(TouchableOpacity));

    expect(checked).toBe(true);
  });

  it('checking of value reverse', () => {
    let checked: boolean = true;
    const onChangeValue = (changed: boolean) => {
      checked = changed;
    };

    const component: RenderAPI = renderComponent({
      checked: checked,
      onChange: onChangeValue,
    });

    fireEvent.press(component.getByType(TouchableOpacity));

    expect(checked).toBe(false);
  });

});
