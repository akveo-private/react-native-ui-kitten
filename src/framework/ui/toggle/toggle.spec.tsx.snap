// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@toggle: matches snapshot active 1`] = `
<View
  appearance="default"
  disabled={false}
  dispatch={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#3366FF",
      "borderRadius": 16,
      "borderWidth": 1.5,
      "height": 32,
      "highlight": Object {
        "backgroundColor": "#e5e7ea",
        "borderRadius": 21,
        "height": 42,
        "width": 62,
      },
      "offsetValue": 20,
      "thumb": Object {
        "backgroundColor": "#ffffff",
      },
      "tintColor": "#DDE1EB",
      "width": 52,
    }
  }
  value={false}
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "position": "absolute",
        },
        Object {
          "backgroundColor": "#e5e7ea",
          "borderRadius": 21,
          "height": 42,
          "width": 62,
        },
      ]
    }
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
  >
    <AnimatedComponent
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#3366FF",
            "borderRadius": 16,
            "borderWidth": 1.5,
            "height": 32,
            "offsetValue": 20,
            "tintColor": "#DDE1EB",
            "width": 52,
          },
          Object {
            "backgroundColor": "rgba(221, 225, 235, 1)",
          },
        ]
      }
    >
      <AnimatedComponent
        style={
          Array [
            Object {
              "alignSelf": "center",
              "position": "absolute",
            },
            Object {
              "borderRadius": 14.5,
              "height": 29,
              "width": 49,
            },
            Object {
              "backgroundColor": "rgba(221, 225, 235, 1)",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            },
          ]
        }
      />
      <AnimatedComponent
        style={
          Array [
            Object {
              "borderRadius": 14.5,
              "borderWidth": 1.5,
              "height": 29,
              "marginHorizontal": 1.5,
              "width": 29,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(221, 225, 235, 1)",
              "elevation": 5,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 29,
            },
          ]
        }
      >
        <CheckMark
          isAnimated={true}
          style={
            Array [
              Object {
                "height": 14.5,
                "width": 14.5,
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
              },
            ]
          }
        />
      </AnimatedComponent>
    </AnimatedComponent>
  </TouchableOpacity>
</View>
`;

exports[`@toggle: matches snapshot active checked 1`] = `
<View
  appearance="default"
  disabled={false}
  dispatch={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#3366FF",
      "borderRadius": 16,
      "borderWidth": 1.5,
      "height": 32,
      "highlight": Object {
        "backgroundColor": "#e5e7ea",
        "borderRadius": 21,
        "height": 42,
        "width": 62,
      },
      "offsetValue": 20,
      "thumb": Object {
        "backgroundColor": "#ffffff",
      },
      "tintColor": "#DDE1EB",
      "width": 52,
    }
  }
  value={true}
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "position": "absolute",
        },
        Object {
          "backgroundColor": "#e5e7ea",
          "borderRadius": 21,
          "height": 42,
          "width": 62,
        },
      ]
    }
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
  >
    <AnimatedComponent
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#3366FF",
            "borderRadius": 16,
            "borderWidth": 1.5,
            "height": 32,
            "offsetValue": 20,
            "tintColor": "#DDE1EB",
            "width": 52,
          },
          Object {
            "backgroundColor": "rgba(51, 102, 255, 1)",
          },
        ]
      }
    >
      <AnimatedComponent
        style={
          Array [
            Object {
              "alignSelf": "center",
              "position": "absolute",
            },
            Object {
              "borderRadius": 14.5,
              "height": 29,
              "width": 49,
            },
            Object {
              "backgroundColor": "rgba(51, 102, 255, 1)",
              "transform": Array [
                Object {
                  "scale": 0.010000000000000009,
                },
              ],
            },
          ]
        }
      />
      <AnimatedComponent
        style={
          Array [
            Object {
              "borderRadius": 14.5,
              "borderWidth": 1.5,
              "height": 29,
              "marginHorizontal": 1.5,
              "width": 29,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "alignSelf": "flex-end",
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(51, 102, 255, 1)",
              "elevation": 5,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 29,
            },
          ]
        }
      >
        <CheckMark
          isAnimated={true}
          style={
            Array [
              Object {
                "height": 14.5,
                "width": 14.5,
              },
              Object {
                "backgroundColor": "rgba(51, 102, 255, 1)",
              },
            ]
          }
        />
      </AnimatedComponent>
    </AnimatedComponent>
  </TouchableOpacity>
</View>
`;

exports[`@toggle: matches snapshot active checked: checked 1`] = `
<View
  appearance="default"
  disabled={false}
  dispatch={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#3366FF",
      "borderRadius": 16,
      "borderWidth": 1.5,
      "height": 32,
      "highlight": Object {
        "backgroundColor": "transparent",
        "borderRadius": 21,
        "height": 42,
        "width": 62,
      },
      "offsetValue": 20,
      "thumb": Object {
        "backgroundColor": "#ffffff",
      },
      "tintColor": "#DDE1EB",
      "width": 52,
    }
  }
  value={true}
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "position": "absolute",
        },
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 21,
          "height": 42,
          "width": 62,
        },
      ]
    }
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
  >
    <AnimatedComponent
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#3366FF",
            "borderRadius": 16,
            "borderWidth": 1.5,
            "height": 32,
            "offsetValue": 20,
            "tintColor": "#DDE1EB",
            "width": 52,
          },
          Object {
            "backgroundColor": "rgba(51, 102, 255, 1)",
          },
        ]
      }
    >
      <AnimatedComponent
        style={
          Array [
            Object {
              "alignSelf": "center",
              "position": "absolute",
            },
            Object {
              "borderRadius": 14.5,
              "height": 29,
              "width": 49,
            },
            Object {
              "backgroundColor": "rgba(51, 102, 255, 1)",
              "transform": Array [
                Object {
                  "scale": 0.010000000000000009,
                },
              ],
            },
          ]
        }
      />
      <AnimatedComponent
        style={
          Array [
            Object {
              "borderRadius": 14.5,
              "borderWidth": 1.5,
              "height": 29,
              "marginHorizontal": 1.5,
              "width": 29,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "alignSelf": "flex-end",
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(51, 102, 255, 1)",
              "elevation": 5,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 29,
            },
          ]
        }
      >
        <CheckMark
          isAnimated={true}
          style={
            Array [
              Object {
                "height": 14.5,
                "width": 14.5,
              },
              Object {
                "backgroundColor": "rgba(51, 102, 255, 1)",
              },
            ]
          }
        />
      </AnimatedComponent>
    </AnimatedComponent>
  </TouchableOpacity>
</View>
`;

exports[`@toggle: matches snapshot active: default 1`] = `
<View
  appearance="default"
  disabled={false}
  dispatch={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#3366FF",
      "borderRadius": 16,
      "borderWidth": 1.5,
      "height": 32,
      "highlight": Object {
        "backgroundColor": "transparent",
        "borderRadius": 21,
        "height": 42,
        "width": 62,
      },
      "offsetValue": 20,
      "thumb": Object {
        "backgroundColor": "#ffffff",
      },
      "tintColor": "#DDE1EB",
      "width": 52,
    }
  }
  value={false}
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "position": "absolute",
        },
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 21,
          "height": 42,
          "width": 62,
        },
      ]
    }
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
  >
    <AnimatedComponent
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#3366FF",
            "borderRadius": 16,
            "borderWidth": 1.5,
            "height": 32,
            "offsetValue": 20,
            "tintColor": "#DDE1EB",
            "width": 52,
          },
          Object {
            "backgroundColor": "rgba(221, 225, 235, 1)",
          },
        ]
      }
    >
      <AnimatedComponent
        style={
          Array [
            Object {
              "alignSelf": "center",
              "position": "absolute",
            },
            Object {
              "borderRadius": 14.5,
              "height": 29,
              "width": 49,
            },
            Object {
              "backgroundColor": "rgba(221, 225, 235, 1)",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            },
          ]
        }
      />
      <AnimatedComponent
        style={
          Array [
            Object {
              "borderRadius": 14.5,
              "borderWidth": 1.5,
              "height": 29,
              "marginHorizontal": 1.5,
              "width": 29,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(221, 225, 235, 1)",
              "elevation": 5,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 29,
            },
          ]
        }
      >
        <CheckMark
          isAnimated={true}
          style={
            Array [
              Object {
                "height": 14.5,
                "width": 14.5,
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
              },
            ]
          }
        />
      </AnimatedComponent>
    </AnimatedComponent>
  </TouchableOpacity>
</View>
`;

exports[`@toggle: matches snapshot checked 1`] = `
<View
  appearance="default"
  disabled={false}
  dispatch={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#3366FF",
      "borderRadius": 16,
      "borderWidth": 1.5,
      "height": 32,
      "highlight": Object {
        "backgroundColor": "transparent",
        "borderRadius": 21,
        "height": 42,
        "width": 62,
      },
      "offsetValue": 20,
      "thumb": Object {
        "backgroundColor": "#ffffff",
      },
      "tintColor": "#DDE1EB",
      "width": 52,
    }
  }
  value={true}
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "position": "absolute",
        },
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 21,
          "height": 42,
          "width": 62,
        },
      ]
    }
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
  >
    <AnimatedComponent
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#3366FF",
            "borderRadius": 16,
            "borderWidth": 1.5,
            "height": 32,
            "offsetValue": 20,
            "tintColor": "#DDE1EB",
            "width": 52,
          },
          Object {
            "backgroundColor": "rgba(51, 102, 255, 1)",
          },
        ]
      }
    >
      <AnimatedComponent
        style={
          Array [
            Object {
              "alignSelf": "center",
              "position": "absolute",
            },
            Object {
              "borderRadius": 14.5,
              "height": 29,
              "width": 49,
            },
            Object {
              "backgroundColor": "rgba(51, 102, 255, 1)",
              "transform": Array [
                Object {
                  "scale": 0.010000000000000009,
                },
              ],
            },
          ]
        }
      />
      <AnimatedComponent
        style={
          Array [
            Object {
              "borderRadius": 14.5,
              "borderWidth": 1.5,
              "height": 29,
              "marginHorizontal": 1.5,
              "width": 29,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "alignSelf": "flex-end",
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(51, 102, 255, 1)",
              "elevation": 5,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 29,
            },
          ]
        }
      >
        <CheckMark
          isAnimated={true}
          style={
            Array [
              Object {
                "height": 14.5,
                "width": 14.5,
              },
              Object {
                "backgroundColor": "rgba(51, 102, 255, 1)",
              },
            ]
          }
        />
      </AnimatedComponent>
    </AnimatedComponent>
  </TouchableOpacity>
</View>
`;

exports[`@toggle: matches snapshot checked disabled 1`] = `
<View
  appearance="default"
  disabled={true}
  dispatch={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#8992A3",
      "borderRadius": 16,
      "borderWidth": 1.5,
      "height": 32,
      "highlight": Object {
        "backgroundColor": "transparent",
        "borderRadius": 21,
        "height": 42,
        "width": 62,
      },
      "offsetValue": 20,
      "thumb": Object {
        "backgroundColor": "#ffffff",
      },
      "tintColor": "#DDE1EB",
      "width": 52,
    }
  }
  value={true}
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
  >
    <AnimatedComponent
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#8992A3",
            "borderRadius": 16,
            "borderWidth": 1.5,
            "height": 32,
            "offsetValue": 20,
            "tintColor": "#DDE1EB",
            "width": 52,
          },
          Object {
            "backgroundColor": "rgba(137, 146, 163, 1)",
          },
        ]
      }
    >
      <AnimatedComponent
        style={
          Array [
            Object {
              "alignSelf": "center",
              "position": "absolute",
            },
            Object {
              "borderRadius": 14.5,
              "height": 29,
              "width": 49,
            },
            Object {
              "backgroundColor": "rgba(137, 146, 163, 1)",
              "transform": Array [
                Object {
                  "scale": 0.010000000000000009,
                },
              ],
            },
          ]
        }
      />
      <AnimatedComponent
        style={
          Array [
            Object {
              "borderRadius": 14.5,
              "borderWidth": 1.5,
              "height": 29,
              "marginHorizontal": 1.5,
              "width": 29,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "alignSelf": "flex-end",
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(137, 146, 163, 1)",
              "elevation": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 29,
            },
          ]
        }
      >
        <CheckMark
          isAnimated={true}
          style={
            Array [
              Object {
                "height": 14.5,
                "width": 14.5,
              },
              Object {
                "backgroundColor": "rgba(137, 146, 163, 1)",
              },
            ]
          }
        />
      </AnimatedComponent>
    </AnimatedComponent>
  </TouchableOpacity>
</View>
`;

exports[`@toggle: matches snapshot default 1`] = `
<View
  appearance="default"
  disabled={false}
  dispatch={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#3366FF",
      "borderRadius": 16,
      "borderWidth": 1.5,
      "height": 32,
      "highlight": Object {
        "backgroundColor": "transparent",
        "borderRadius": 21,
        "height": 42,
        "width": 62,
      },
      "offsetValue": 20,
      "thumb": Object {
        "backgroundColor": "#ffffff",
      },
      "tintColor": "#DDE1EB",
      "width": 52,
    }
  }
  value={false}
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "position": "absolute",
        },
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 21,
          "height": 42,
          "width": 62,
        },
      ]
    }
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
  >
    <AnimatedComponent
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#3366FF",
            "borderRadius": 16,
            "borderWidth": 1.5,
            "height": 32,
            "offsetValue": 20,
            "tintColor": "#DDE1EB",
            "width": 52,
          },
          Object {
            "backgroundColor": "rgba(221, 225, 235, 1)",
          },
        ]
      }
    >
      <AnimatedComponent
        style={
          Array [
            Object {
              "alignSelf": "center",
              "position": "absolute",
            },
            Object {
              "borderRadius": 14.5,
              "height": 29,
              "width": 49,
            },
            Object {
              "backgroundColor": "rgba(221, 225, 235, 1)",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            },
          ]
        }
      />
      <AnimatedComponent
        style={
          Array [
            Object {
              "borderRadius": 14.5,
              "borderWidth": 1.5,
              "height": 29,
              "marginHorizontal": 1.5,
              "width": 29,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(221, 225, 235, 1)",
              "elevation": 5,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 29,
            },
          ]
        }
      >
        <CheckMark
          isAnimated={true}
          style={
            Array [
              Object {
                "height": 14.5,
                "width": 14.5,
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
              },
            ]
          }
        />
      </AnimatedComponent>
    </AnimatedComponent>
  </TouchableOpacity>
</View>
`;

exports[`@toggle: matches snapshot disabled 1`] = `
<View
  appearance="default"
  disabled={true}
  dispatch={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 52,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "borderColor": "#8992A3",
      "borderRadius": 16,
      "borderWidth": 1.5,
      "height": 32,
      "highlight": Object {
        "backgroundColor": "transparent",
        "borderRadius": 21,
        "height": 42,
        "width": 62,
      },
      "offsetValue": 20,
      "thumb": Object {
        "backgroundColor": "#ffffff",
      },
      "tintColor": "#DDE1EB",
      "width": 52,
    }
  }
  value={false}
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
  >
    <AnimatedComponent
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#8992A3",
            "borderRadius": 16,
            "borderWidth": 1.5,
            "height": 32,
            "offsetValue": 20,
            "tintColor": "#DDE1EB",
            "width": 52,
          },
          Object {
            "backgroundColor": "rgba(221, 225, 235, 1)",
          },
        ]
      }
    >
      <AnimatedComponent
        style={
          Array [
            Object {
              "alignSelf": "center",
              "position": "absolute",
            },
            Object {
              "borderRadius": 14.5,
              "height": 29,
              "width": 49,
            },
            Object {
              "backgroundColor": "rgba(221, 225, 235, 1)",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            },
          ]
        }
      />
      <AnimatedComponent
        style={
          Array [
            Object {
              "borderRadius": 14.5,
              "borderWidth": 1.5,
              "height": 29,
              "marginHorizontal": 1.5,
              "width": 29,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(221, 225, 235, 1)",
              "elevation": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 29,
            },
          ]
        }
      >
        <CheckMark
          isAnimated={true}
          style={
            Array [
              Object {
                "height": 14.5,
                "width": 14.5,
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
              },
            ]
          }
        />
      </AnimatedComponent>
    </AnimatedComponent>
  </TouchableOpacity>
</View>
`;
